R/mle-finder.R

```diff
@@ -21,3 +21,76 @@ linear_BFGS <- function(design, outcome, noise_var = 1) {
   return(MLE)
 }
 
+logit_loglik <- function(coef, design, outcome) {
+
+  n <- length(outcome)
+  log_lik <- sum(sapply(1:n, function(i){
+    outcome[i] * sum(design[i,] * coef) -
+      log(1 + exp(sum(design[i,] * coef)))
+    }))
+  return(log_lik)
+
+}
+
+logit_gradient <- function(coef, design, outcome) {
+
+  pi <- exp(design %*% coef) / (1 + exp(design %*% coef))
+  gradient <- as.vector(t(outcome - pi) %*% design)
+
+  return(gradient)
+
+}
+
+logit_BFGS <- function(design, outcome) {
+  initial <- rep(0, ncol(design))
+  MLE <- stats::optim(initial, logit_loglik, logit_gradient,
+                      design = design, outcome = outcome,
+                      control = list(fnscale = -1),
+                      method = "BFGS"
+  )$par
+  return(MLE)
+}
+
+
+logit_hessian <- function(coef, design, outcome){
+  eta <- design %*% coef
+  pi <- exp(eta)/(1+exp(eta))
+  W <- diag(c(pi * (1 - pi)))
+  hessian <- t(design) %*% W %*% design
+  return(hessian)
+}
+
+logit_newton <- function(design, outcome){
+
+  max_iteration <- 1000
+  t <- 0
```

Comment:
Use a more informative name here, like `iters`. Even `i` might be more intuitive for most programmers than `t`.